Sources/Commands/SwiftTestTool.swift

```diff
@@ -78,11 +78,39 @@ struct SharedOptions: ParsableArguments {
           help: "Enable support for XCTest")
     var enableXCTestSupport: Bool = true
 
-    /// Whether to enable support for swift-testing.
+    /// Storage for whether to enable support for swift-testing.
     @Flag(name: .customLong("experimental-swift-testing"),
           inversion: .prefixedEnableDisable,
           help: "Enable experimental support for swift-testing")
-    var enableSwiftTestingLibrarySupport: Bool = false
+    var _enableSwiftTestingLibrarySupport: Bool?
+
+    /// Whether to enable support for swift-testing.
+    func enableSwiftTestingLibrarySupport(swiftTool: SwiftTool) throws -> Bool {
+        // Honor the user's explicit command-line selection, if any.
+        if let callerSuppliedValue = _enableSwiftTestingLibrarySupport {
+            return callerSuppliedValue
+        }
+
+        // If the active package has a dependency on swift-testing, automatically enable support for it so that extra steps are not needed.
+        let workspace = try swiftTool.getActiveWorkspace()
+        let root = try swiftTool.getWorkspaceRoot()
+        let rootManifests = try temp_await {
+            workspace.loadRootManifests(
+                packages: root.packages,
+                observabilityScope: swiftTool.observabilityScope,
+                completion: $0
+            )
+        }
+        return rootManifests.values.contains { manifest in
+            manifest.dependencies.contains { dependency in
+                if case let .sourceControl(sourceControl) = dependency,
+                   case let .remote(url) = sourceControl.location {
+                    return url.absoluteString == "https://github.com/apple/swift-testing.git"
```

Comment:
I think we should check for the package identity being "swift-testing" instead and also support path-based dependencies.

Reply:
+1

Reply:
IIUC checking for package identity will take path-based dependencies into account automagically